src/CAN/can_mapping.c

```diff
@@ -32,6 +32,8 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
         }
         uint32_t bitmask = (1UL << length) - 1;
         uint32_t raw_value = (raw_data >> offset) & bitmask;
+
+        /* normalize endian */
         if (mapping->big_endian) {
                 switch (mapping->length) {
```

Comment:
Agree on handling all allowed cases explicitly. 

Are you suggesting a panic(PANIC_CAUSE_UNREACHABLE);  if >4 for length?